src/acl_printf.cpp

```diff
@@ -980,17 +1002,21 @@ void acl_schedule_printf_buffer_pickup(int activation_id, int size,
   }
 
 #ifdef DEBUG
-  printf("printf pickup %d %d %d\n", activation_id, size, stalled);
+  printf("printf pickup %d %d\n", activation_id, size);
   fflush(stdout);
 #endif
   if (activation_id >= 0 && activation_id < doq->max_ops) {
     // This address is stable, given a fixed activation_id.
     // So we don't run into race conditions.
     acl_device_op_t *op = doq->op + activation_id;
-
-    stalled = stalled; // this argument is no longer used!
-
     op->info.num_printf_bytes_pending = (cl_uint)size;
+
+    // Propagate the operation info
+    if (debug_dump_printf == 1) {
+      op->info.debug_dump_printf = 1;
+    } else {
+      op->info.debug_dump_printf = 0;
+    }
```

Comment:
You could shorten to
```
op->info.debug_dump_printf = debug_dump_printf ? 1 : 0;
```
i.e., zero input value zero yields 0 and non-zero input value is normalized to 1.

Reply:
Thanks Peter. Your suggestion looks cleaner and handled any other non-zero value.